src/ipld.rs

```diff
@@ -66,7 +86,55 @@ impl fmt::Debug for Ipld {
     }
 }
 
-/// An index into ipld
+/// IPLD Kind information without the actual value.
+///
+/// Sometimes it's useful to know the kind of an Ipld object without the actual value, e.g. for
+/// error reporting. Those kinds can be a unity-only enum.
+#[derive(Clone, Debug)]
+pub enum IpldKind {
+    /// Null type.
+    Null,
+    /// Boolean type.
+    Bool,
+    /// Integer type.
+    Integer,
+    /// Float type.
+    Float,
+    /// String type.
+    String,
+    /// Bytes type.
+    Bytes,
+    /// List type.
+    List,
+    /// Map type.
+    Map,
+    /// Link type.
+    Link,
+}
+
+impl IpldKind {
+    /// Convert from an [`Ipld`] object into its kind without any associated values.
+    ///
+    /// This is intentionally not implemented via `From<Ipld>` to prevent accidental conversions by
+    /// making it more explicit.
+    pub fn from_ipld(ipld: &Ipld) -> Self {
```

Comment:
So... in retrospect, maybe `fn kind(&self)` on `Ipld` instead?

Reply:
Interesting, I had the same thought when I was coding this and was undecided, so I went for your proposal. Looks like I'll switch :)